testcases/kernel/syscalls/fallocate/fallocate03.c

```diff
@@ -1,277 +1,94 @@
- *
- *
- *	INPUT SPECIFICATIONS
- *		No input needs to be specified
- *		  fallocate() in-puts are specified through test_data
- *
- *	OUTPUT SPECIFICATIONS
- *		Output describing whether test cases passed or failed.
- *
- *	ENVIRONMENTAL NEEDS
- *		Test Needs to be executed on file system supporting ext4
- *   LTP {TMP} Needs to be set to such a folder
- *
- *	SPECIAL PROCEDURAL REQUIREMENTS
- *		None
- *
- *	DETAILED DESCRIPTION
- *		This is a test case for fallocate() system call.
- *		This test suite tests working of fallocate on sparse file
- *		fallocate is tested for different offsets
- *
- *		Total 8 Test Cases :-
- *			Different offsets with in a sparse file is tested
- *
- *	Setup:
- *		Setup file on which fallocate is to be called
- *		Set up a file with hole, created through lseek
- *
- *	Test:
- *		Loop if the proper options are given
- *		Execute system call
- *		Check return code, if system call failed
- *		TEST fails, PASS the test otherwise
- *
- *	Cleanup:
- *		Cleanup the temporary folder
- *
-*************************************************************************/
+/*\
+ * [Description]
+ * Test fallocate() on sparse file for different offsets, with a total of 8 test cases
+ */
 
 #define _GNU_SOURCE
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <endian.h>
-#include <errno.h>
-#include <sys/stat.h>
-#include <sys/types.h>		//Can be done with out
 #include <fcntl.h>
-#include <unistd.h>
-#include <inttypes.h>
-#include <sys/utsname.h>
-
-#include "test.h"
-#include "safe_macros.h"
-#include "lapi/fallocate.h"
+#include "tst_test.h"
 
 #define BLOCKS_WRITTEN 12
 #define HOLE_SIZE_IN_BLOCKS 12
 #define DEFAULT_MODE 0
-#define TRUE 0
-
-void get_blocksize(int);
-void populate_file();
-void file_seek(off_t);
 
-char *TCID = "fallocate03";
-char fname[255];
-int fd;
-struct test_data_t {
+static int fd;
+static struct test_case {
 	int mode;
 	loff_t offset;
-	loff_t len;
-	int error;
-} test_data[] = {
-	{
-	DEFAULT_MODE, 2, 1, TRUE}, {
-	DEFAULT_MODE, BLOCKS_WRITTEN, 1, TRUE}, {
-	DEFAULT_MODE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS / 2 - 1, 1, TRUE},
-	{
-	DEFAULT_MODE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 1, 1, TRUE}, {
-	FALLOC_FL_KEEP_SIZE, 2, 1, TRUE}, {
-	FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN, 1, TRUE}, {
-	FALLOC_FL_KEEP_SIZE,
-		    BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS / 2 + 1, 1, TRUE}, {
-	FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 2,
-		    1, TRUE}
+} test_cases[] = {
+	{DEFAULT_MODE, 2},
+	{DEFAULT_MODE, BLOCKS_WRITTEN},
+	{DEFAULT_MODE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS / 2 - 1},
+	{DEFAULT_MODE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 1},
+	{FALLOC_FL_KEEP_SIZE, 2},
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN},
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS / 2 + 1},
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 2}
 };
 
-int TST_TOTAL = sizeof(test_data) / sizeof(test_data[0]);
-int block_size;
-int buf_size;
+static int block_size;
+static int buf_size;
```

Comment:
Am I correct, that there could be just a single variable for these two? Both are just `stat.st_blksize`, right?

Reply:
You're correct, they are the same value but I felt it makes the test a bit more readable to keep them logically separated. I'll prepare a better proposal on the next push :)